### PR TITLE
[JSC] Add more explicit exception checks to objectProtoFuncLookupGetter objectProtoFuncLookupSetter

### PR DESCRIPTION
**Pull request by Constellation**
#### 6e08c53f024309fbd2ab8dd72a14828d022b7803
<pre>
[JSC] Add more explicit exception checks to objectProtoFuncLookupGetter objectProtoFuncLookupSetter
<a href="https://bugs.webkit.org/show_bug.cgi?id=245304">https://bugs.webkit.org/show_bug.cgi?id=245304</a>
&lt;rdar://98498151&gt;

Reviewed by Mark Lam.

getOwnPropertyDescriptor call was missing explicit exception check before. It was ok because we are checking the result
value. But let&apos;s just deploy simpler way here and suppress failures in debug build.

* Source/JavaScriptCore/runtime/ObjectPrototype.cpp:
(JSC::JSC_DEFINE_HOST_FUNCTION):

Canonical link: <a href="https://commits.webkit.org/254581@main">https://commits.webkit.org/254581@main</a>
</pre>

https://github.com/WebKit/WebKit/commit/fbf31cb6cfb92575fe878b8e3e54ede7cbc6f937

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/89519 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/69/builds/34071 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/20240 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/98824 "Built successfully") | [✅ 🛠 🧪 win](https://ews-build.webkit.org/#/builders/10/builds/155131 "Built successfully and passed tests") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/64/builds/32565 "Built successfully") | [  ~~🛠 mac-debug~~](https://ews-build.webkit.org/#/builders/71/builds/28047 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/81887 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/93235 "Built successfully") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/95166 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/68/builds/25869 "Passed tests") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/76397 "Built successfully") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/35/builds/25814 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/9/builds/80747 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/3/builds/80632 "Passed tests") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/34/builds/68806 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/1/builds/81192 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/66/builds/30323 "Built successfully") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/73/builds/14708 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/38/builds/75015 "Built successfully") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/67/builds/30070 "Built successfully") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/70/builds/15643 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/46/builds/26405 "Passed tests") | 
| [❌ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/74/builds/3219 "Failed to push commit to Webkit repository") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/63/builds/33518 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/60/builds/38599 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🛠 jsc-mips](https://ews-build.webkit.org/#/builders/37/builds/77883 "Built successfully") | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/65/builds/32228 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/34734 "Passed tests") | [✅ 🧪 jsc-mips-tests](https://ews-build.webkit.org/#/builders/45/builds/17243 "Passed tests") |

### CODE REVIEW

File: Source/JavaScriptCore/runtime/ObjectPrototype.cpp [hunk 1]
```diff
@@ -225,7 +225,9 @@ JSC_DEFINE_HOST_FUNCTION(objectProtoFuncLookupGetter, (JSGlobalObject* globalObj
         if (slot.attributes() & PropertyAttribute::CustomAccessor) {
             PropertyDescriptor descriptor;
             ASSERT(slot.slotBase());
-            if (slot.slotBase()->getOwnPropertyDescriptor(globalObject, propertyName, descriptor))
+            bool result = slot.slotBase()->getOwnPropertyDescriptor(globalObject, propertyName, descriptor);
+            RETURN_IF_EXCEPTION(scope, { });
```
Comment (MenloDorian) [inline Source/JavaScriptCore/runtime/ObjectPrototype.cpp]: You can do this instead:
```
EXCEPTION_ASSERT(!scope.exception() || !result);
```

File: Source/JavaScriptCore/runtime/ObjectPrototype.cpp [hunk 2]
```diff
@@ -255,7 +257,9 @@ JSC_DEFINE_HOST_FUNCTION(objectProtoFuncLookupSetter, (JSGlobalObject* globalObj
...
```
Comment (MenloDorian) [inline Source/JavaScriptCore/runtime/ObjectPrototype.cpp]: ditto.

File: Source/JavaScriptCore/runtime/ObjectPrototype.cpp [hunk 3]
```diff
@@ -239,14 +241,14 @@ JSC_DEFINE_HOST_FUNCTION(objectProtoFuncLookupSetter, (JSGlobalObject* globalObj
...
+    RETURN_IF_EXCEPTION(scope, { });
 
     auto propertyName = callFrame->argument(0).toPropertyKey(globalObject);
-    RETURN_IF_EXCEPTION(scope, encodedJSValue());
+    RETURN_IF_EXCEPTION(scope, { });
 
     PropertySlot slot(thisObject, PropertySlot::InternalMethodType::GetOwnProperty);
     bool hasProperty = thisObject->getPropertySlot(globalObject, propertyName, slot);
-    EXCEPTION_ASSERT(!scope.exception() || !hasProperty);
+    RETURN_IF_EXCEPTION(scope, { });
```
Comment (MenloDorian) [inline Source/JavaScriptCore/runtime/ObjectPrototype.cpp]: Why this change?

File: Source/JavaScriptCore/runtime/ObjectPrototype.cpp [hunk 4]
```diff
@@ -209,14 +209,14 @@ JSC_DEFINE_HOST_FUNCTION(objectProtoFuncLookupGetter, (JSGlobalObject* globalObj
...
```
Comment (MenloDorian) [inline Source/JavaScriptCore/runtime/ObjectPrototype.cpp]: Why this change?

**Review by MenloDorian — APPROVED** [review]
r=me

File: Source/JavaScriptCore/runtime/ObjectPrototype.cpp (hunk 4 above)
Comment (Constellation) [reply Source/JavaScriptCore/runtime/ObjectPrototype.cpp]: Just for simplification, but reverted.

File: Source/JavaScriptCore/runtime/ObjectPrototype.cpp (hunk 3 above)
Comment (Constellation) [reply Source/JavaScriptCore/runtime/ObjectPrototype.cpp]: Ditto.

File: Source/JavaScriptCore/runtime/ObjectPrototype.cpp (hunk 2 above)
Comment (Constellation) [reply Source/JavaScriptCore/runtime/ObjectPrototype.cpp]: Changed.

File: Source/JavaScriptCore/runtime/ObjectPrototype.cpp (hunk 1 above)
Comment (Constellation) [reply Source/JavaScriptCore/runtime/ObjectPrototype.cpp]: Changed.
